File: array/0674_longest_continuous_increasing_subsequence/0674_longest_continuous_increasing_subsequence.go
package main

import (
	"fmt"
	"math"
)

func findLengthOfLCIS(nums []int) int {
	res, cur := 0, 0
	for i := 0; i < len(nums); i++ {
		if i == 0 || nums[i-1] < nums[i] {
			cur++
		} else {
			res = int(math.Max(float64(res), float64(cur)))
			cur = 1
		}
	}
	return int(math.Max(float64(res), float64(cur)))
}

func main() {
	nums := []int{1, 3, 5, 4, 7}
	fmt.Println(findLengthOfLCIS(nums))
}

File: array/0674_longest_continuous_increasing_subsequence/0674_longest_continuous_increasing_subsequence.py
# -*- coding: utf-8 -*-


class Solution(object):

    def findLengthOfLCIS(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        result, count = 0, 0
        for i in xrange(len(nums)):
            if i == 0 or nums[i - 1] < nums[i]:
                count += 1
                result = max(result, count)
            else:
                count = 1
        return result


print(Solution().findLengthOfLCIS([1, 3, 5, 4, 7]))